AuroraEditor/Utils/AuroraEditorConfig.swift

```diff
@@ -0,0 +1,212 @@
+        ]
+    ]
+
+    init(fromPath: String) {
+        if let configFile = findEditorConfig(fromPath: fromPath),
+           let configData = FileManager.default.contents(atPath: configFile),
+           let configINI = String(data: configData, encoding: .utf8) {
+            let parsed = AuroraINIParser(ini: configINI).parse()
+            Log.info("INI=\(configFile)")
+            Log.info(parsed)
+            self.parsed = parsed
+        }
+    }
+
+    /// Get value for type
+    /// - Parameters:
+    ///   - value: value type
+    ///   - for: for type
+    public func get(value: EditorConfigKeys, for file: String = "*") -> String {
+        if let value = getKeyNameFor(parsed, file: file)?[value.rawValue] as? String {
+            return value
+        }
+
+        if let value = parsed?["*"]?[value.rawValue] as? String {
+            return value
+        }
+
+        if let value = getKeyNameFor(defaults, file: file)?[value.rawValue] as? String {
+            return value
+        }
+
+        if let value = defaults["*"]?[value.rawValue] as? String {
+            return value
+        }
+
+        Log.error("There is no value for \(value.rawValue) for file \(file)")
+        return ""
+    }
+
+    private func shouldInvert(_ value: Bool, inverted: Bool) -> Bool {
+        return inverted ? !value : value
+    }
+
+    private func getKeyNameFor(_ inDict: [String: [String: Any]]?, file: String) -> [String: Any]? {
+        var invert = false
+
+        if let inDict = inDict {
+            for (pattern, values) in inDict {
+                if pattern.hasPrefix("!") {
+                    invert = true
+                }
+
+                if pattern.contains("**") {
+                    Log.info("** is not (yet) supported")
+                    continue
+                } else if pattern.hasPrefix("*.") {
+                    let search = pattern.replacingOccurrences(of: "*.", with: "")
+
+                    if pattern.contains("{") && pattern.contains("}") {
+                        if let noOpen = pattern.split(separator: "{").last,
+                           let noClose = noOpen.split(separator: "}").first {
+                            var extensionList = noClose.split(separator: ",")
+
+                            for fileExtension in extensionList where shouldInvert(
+                                file.hasSuffix(fileExtension),
+                                inverted: invert
+                            ) {
+                                return values
+                            }
+                        }
+                    }
+
+                    if file.hasSuffix(search) {
+                        return values
+                    }
+                } else {
+                    if file.hasSuffix(pattern) {
+                        return values
+                    }
+                }
+            }
+        }
+
+        return nil
+    }
+
+    private func findEditorConfig(fromPath: String) -> String? {
```

Comment:
Maybe find the config file in the Aurora Editor folder, it would follow a pretty standard experience across all known editors too. Also if create the the config file we should automatically add the config directory to the projects ".gitignore" the Source Control Kit should have this functionality of adding it. If not I'll look into it.

Reply:
We should NOT do that.
https://editorconfig.org/

EditorConfig helps maintain consistent coding styles for multiple developers working on the same project across various editors and IDEs. The EditorConfig project consists of a file format for defining coding styles and a collection of text editor plugins that enable editors to read the file format and adhere to defined styles. EditorConfig files are easily readable and they work nicely with version control systems.